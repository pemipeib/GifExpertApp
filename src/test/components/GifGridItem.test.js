import React from 'react';
import {shallow} from "enzyme";

import GifGridItem from "../../components/GifGridItem";

const title = 'Un titulo';
const url = 'http://localhost/algo.jpg';
const wrapper = shallow(
    <GifGridItem
        title={title}
        url={url}
    />
);

describe('Pruebas en <GifGridItem />', () =>{
    
    test('Debe mostrar <GifGridItem /> correctamente', () => {
                const wrapper = shallow(
                    <GifGridItem
                        title={title}
                        url={url}
                    />
                );
        expect(wrapper).toMatchSnapshot();
    });
    
    test('Debe mostrar el title en el pÃ¡rrafo', () => {
        const p = wrapper.find('p')
        expect(p.text().trim()).toBe(title);
    });
    
    test('Debe tener imagen con la url y el alt', () => {
        const img = wrapper.find('img')
        expect(img.prop('src')).toBe(url); //prop de enzyme llama directamente a la propiedad de la etiqueta
        expect(img.prop('alt')).toBe(title);
    });
    
    test('Debe tener la clase animate__backInDown', () => {
        const div = wrapper.find('div');
        expect(div.prop('className')).toContain('animate__backInDown');
    });
    
});
