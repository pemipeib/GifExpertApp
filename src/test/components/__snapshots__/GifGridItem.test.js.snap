// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <GifGridItem /> Debe mostrar <GifGridItem /> correctamente 1`] = `
<div
  className="card animate__animated animate__backInDown"
>
  <img
    alt="Un titulo"
    src="http://localhost/algo.jpg"
  />
  <p>
    Un titulo
  </p>
</div>
`;
